File: src/ssr_pkg_p/script/chao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from my_msgs_p.msg import mymsg
if __name__ == "__main__":
    rospy.init_node("chao_node")
    rospy.logwarn("Hello ROS! This is chao_node.")

    pub=rospy.Publisher("chao_msg",mymsg,queue_size=10)
    rate=rospy.Rate(10)
    while not rospy.is_shutdown():
        rospy.loginfo("This is Chao_node.")
        msg=mymsg()
        msg.grade = "King"
        msg.star = 50
        msg.data = "This is chao_node sending messages."
        pub.publish(msg)
        rate.sleep()

File: src/ssr_pkg/src/chao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char *argv[])
{
    ros::init(argc,argv,"chao_node");
    printf("Hello ROS!\n");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<std_msgs::String>("chao_msg",10);

    std_msgs::String msg;

    ros::Rate loop_rate(10);
    while (ros::ok)
    {
        printf("Keep on!\n");
        msg.data = "This is Chao sensor\r\n";
        pub.publish(msg);
        loop_rate.sleep();
    }
    
    return 0;
}
